### PR TITLE
python3Packages.gocardless-pro: init at 1.45.0

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

An official client for the GoCardless Pro API.

Package used by `frappe/erpnext` (packaging ongoing).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 243891` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.gocardless-pro</li>
    <li>python311Packages.gocardless-pro</li>
  </ul>
</details>

**Review by natsukium — COMMENTED** [review]
Would you please use `nixpkgs-fmt` for formatting?
Besides this, LGTM.

**Comment by blaggacao** [comment]
>  nixpkgs-fmt for formatting

Copy-paste oversight from a different repo with different policy, fixed in [force-push](https://github.com/NixOS/nixpkgs/compare/19a737705b468aea8a56dbeb4f53d4474f4df19d..939ee5ea3bbef68d20507a35b0b4fd33128258a7). Thx!

File: pkgs/development/python-modules/gocardless-pro/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    responses
+    nose
+  ];
+
+  meta = with lib; {
+    description = "A client library for the GoCardless Pro API";
+    homepage = "https://pypi.org/project/gocardless-pro/";
```
Comment (natsukium) [inline pkgs/development/python-modules/gocardless-pro/default.nix:38]:
```suggestion
    homepage = "https://github.com/gocardless/gocardless-pro-python";
```
Wouldn't github be better for the homepage?
The source is also fetched from GitHub.

File: pkgs/development/python-modules/gocardless-pro/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, buildPythonPackage
+, fetchFromGitHub
+, requests
+, six
+, pytestCheckHook
+, responses
+, nose
+,
+}:
+buildPythonPackage rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/gocardless-pro/default.nix:11]:
```suggestion
}:

buildPythonPackage rec {
```
This is a common pitfall when switching from alejandra to nixpkgs-fmt.

File: pkgs/development/python-modules/gocardless-pro/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/python-modules/gocardless-pro/default.nix]: Oh! Absolutely, sry the oversight. :smile:

**Comment by natsukium** [comment]
@ofborg build python310Packages.gocardless-pro

**Review by natsukium — APPROVED** [review]
LGTM

**Review by dotlambda — APPROVED** [review]

**Pull request merged**